Pas 50

BraidenCutforth opened the pull request.

daelynj commented on src/services/calculations/gpaConverter.ts:
```diff
@@ -0,0 +1,18 @@
+    // 3 => C += 2.33, 4.33 scale
+const gpaMatrix = [
+    [4, 4, 3.7, 3.3, 3, 2.7, 2.3, 2, 1, 0],
+    [4, 4,	3.67, 3.33, 3, 2.67, 2.33, 2, 1, 0],
```
Is there supposed to be a tab before `3.67`?

BraidenCutforth replied:
Should not have the tab

daelynj commented on src/services/calculations/gpaConverter.ts:
```diff
@@ -0,0 +1,18 @@
+    [4.33, 4, 3.7, 3.3, 3, 2.7, 2.3, 2, 1, 0],
+    [4.33, 4, 3.67, 3.33, 3, 2.67, 2.33, 2, 1, 0]]
+
+function convert(grade : number, convertTo : number) {
```
smallest nitpick ever, but is there usually supposed to be a space before the colon when defining a type?
`grade : number`
or
`grade: number`

BraidenCutforth replied:
Definitely the `grade: number` style

BraidenCutforth replied:
Also, we can probably do some math to find out the exact 4 point gpa, say someone has 7.5 and is using scheme 1, we can do gpaMatrix[1][7] + (gpaMatrix[1][8]-gpaMatrix[1][7])*0.5. I.e the truncated gpa + the step difference * the 9 point decimal

BraidenCutforth replied:
Also need to export this: `export function ...`

louis-kedziora replied:
Done it now does exact conversions

BraidenCutforth commented on src/services/calculations/gpaConverter.ts:
```diff
@@ -0,0 +1,30 @@
+    description: string
+}
+
+const infos: GpaInfo[] = [
```
Let's export this. And let's rename it to `gpaSchemes` I think

louis-kedziora replied:
Done also exported the convert

BraidenCutforth commented on src/services/calculations/gpaConverter.ts:
```diff
@@ -0,0 +1,30 @@
+    return newArr[Math.trunc(grade)]
+}
+
+interface GpaInfo {
```
I think rename to `GpaScheme`

louis-kedziora replied:
Done

BraidenCutforth committed:
Minor bump for dependency for security fix

BraidenCutforth committed:
PAS-50: Sign Up Page

- Added regular components to sign up page
- Enabled create account redirect button on SignIn page
- Adjusted text link size in signin/up pages

isaiahdoyle commented on src/components/MainActionButton.tsx:
```diff
@@ -12,6 +12,7 @@ export interface mainActionButtonInterface {
 	children: React.ReactNode;
 	onClick: () => void;
 	variant?: Variant;
+	disabled?: boolean;
```
Is there going to be styling for this? Should I add it to the variants? 👀

BraidenCutforth replied:
There will be styling for this and I created a separate ticket. Essentially we will want to make the button look unclickable, but for now, it will still do nothing

seiyaterada commented on src/pages/accounts/SignInPage.tsx:
```diff
@@ -40,24 +41,23 @@ const TextLinkContainer = styled.div`
 	const [emailErr, setEmailErr] = useState(false);
 	const [userPass, setUserPass] = useState('');
 
+	const history = useHistory();
```
What does the `useHistory()` function do from react-router-dom?

BraidenCutforth replied:
So you're definitely familiar with the browser history (the back and forward button). This essentially allows you to access this history through the react-router lib. When you push onto the history, it changes your page and updates the URL. It also allows the user to hit the browser back button to go to the last page they were on, giving the application a multi page feel, while actually being an SPA

seiyaterada commented on src/pages/accounts/SignUpPage.tsx:
```diff
@@ -12,46 +16,35 @@ const SignUpPageContainer = styled.div`
 
 const SignUpContents = styled.div`
-	width: 250px;
+	max-width: 300px;
```
Is this supposed to be using px? Or should it be using rem or em?

BraidenCutforth replied:
So this I actually don't want to vary with font size (which rem and em do). Since this is a top level layout px is fine. And it is meant to be based on a users screen in px.

seiyaterada commented on src/pages/accounts/SignUpPage.tsx:
```diff
@@ -95,93 +90,72 @@ function SignUpPage() {
+							setEmailErr(true);
+					}}
+				/>
+				<TextInput
+					label="Password"
+					required
+					type="password"
+					value={userPass}
+					onChange={(e) => {
+						if (validPass(e.target.value)) setPassErr(false);
+						setUserPass(e.target.value);
```
Is the password being blurred somehow when being inputted (ex. turning the characters into *** or dots or something)? If not should this be a feature?

BraidenCutforth replied:
So there is a special input type password that is native html that will * out the password (`<input type="password />`). As you can see I pass through the type to the input element in the text input.

seiyaterada reviewed (approved):
Overall the code looks great and easy to follow! I just added a few minor comments and questions if you could answer those please :)

isaiahdoyle commented on src/pages/accounts/SignUpPage.tsx:
```diff
@@ -95,93 +90,72 @@ function SignUpPage() {
+					<TextLink
+						text="Already have an account? Sign in here."
+						onClick={() => {
+							history.push('/sign-in');
```
Same question as Seiya; what is `useHistory` for?

BraidenCutforth replied:
As stated in the response to Seiya

So you're definitely familiar with the browser history (the back and forward button). This essentially allows you to access this history through the react-router lib. When you push onto the history, it changes your page and updates the URL. It also allows the user to hit the browser back button to go to the last page they were on, giving the application a multi page feel, while actually being an SPA.

isaiahdoyle reviewed (approved):
Beautiful as always! I just left a couple questions on minor things :^)
